src/ServiceControl/Recoverability/Retrying/Infrastructure/ReturnToSender.cs

```diff
@@ -27,6 +29,7 @@ public virtual async Task HandleMessage(MessageContext message, IDispatchMessage
             var outgoingHeaders = new Dictionary<string, string>(message.Headers);
 
             outgoingHeaders.Remove("ServiceControl.Retry.StagingId");
+            outgoingHeaders["ServiceControl.Retry.AcknowledgementQueue"] = settings.ErrorQueue;
```

Comment:
will that be the correct address for remote endpoints? e.g. when using MSMQ?

Reply:
Great catch @timbussmann ❤️  If we released it as-was it would have been a disaster. We fixed it by doing some `TransportInfrastructure` hackery and I've checked in debug that now it sets proper queue name i.e. `error@LAPTOP-PHIOSCQO`